### PR TITLE
NIFI-6578 - Upgraded zookeeper framework version. Some code changes r…

### PR DESCRIPTION
**Pull request by thenatog**
…equired. This change also required a change to the embedded zookeeper.

NIFI-6578 - Updating tests to account for the new node /zookeeper/config

Upgraded Zookeeper and Curator versions for zookeeper migrator in toolkit. Updated tests to allow for new znode /zookeeper/config

NIFI-6578 - Added documentation changes to reflect changes to the zookeeper.properties file going foward.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by thenatog** [comment]
@markap14 @jtstorck

**Comment by thenatog** [comment]
To test this PR:

With embedded zookeeper:
- nifi.zookeeper.connect.string=localhost:2181
- nifi.state.management.embedded.zookeeper.start=true
- zookeeper.properties file:
    - Remove clientPort=2181
    - Add port to end of server string: server.1=localhost:2888:3888;2181

With external zookeeper quorum:
- Write the docker-compose.yml file from this page: https://hub.docker.com/_/zookeeper to a directory, run docker-compose up 
- Configure NiFi to use the docker zookeeper nodes:
    - nifi.properties: nifi.zookeeper.connect.string=zoo1:2281,zoo2:2282,zoo3:2283
    - state-management.xml:         
<property name="Connect String">zoo1:2281,zoo2:2282,zoo:2283</property>

In zookeeper.properties file define:
- server.1=zoo1:2888:3888;2281 
- server.2=zoo2:2888:3888;2282
-  server.3=zoo32888:3888;2282

File: nifi-toolkit/nifi-toolkit-zookeeper-migrator/pom.xml [hunk 1]
```diff
@@ -78,13 +78,19 @@
...
         </dependency>
         <dependency>
             <groupId>org.apache.curator</groupId>
             <artifactId>curator-client</artifactId>
-            <version>2.11.0</version>
+            <version>4.2.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
```
Comment (markap14) [inline nifi-toolkit/nifi-toolkit-zookeeper-migrator/pom.xml]: I don't believe we need this, do we?

Comment (thenatog) [reply nifi-toolkit/nifi-toolkit-zookeeper-migrator/pom.xml]: Nope, will remove

**Comment by markap14** [comment]
Thanks @thenatog all looks good to me. Merged to master.

**Pull request closed without merging**